src/renderer/modals/Unfreeze/steps/StepAmount.js

```diff
@@ -71,10 +104,34 @@ export default function StepAmount({
 
   const selectEnergy = useCallback(useCallback(() => onChange("ENERGY"), [onChange]));
 
-  const { tronResources: { frozen: { bandwidth, energy } = {} } = {} } = account;
+  const { tronResources } = account;
+
+  const {
+    frozen: { bandwidth, energy },
+  } = tronResources || {};
+
+  /** ! expiredAt should always be set with the amount if not this will disable the field by default ! */
+  const { amount: bandwidthAmount, expiredAt: _bandwidthExpiredAt } = bandwidth || {};
+  const bandwidthExpiredAt = +new Date(_bandwidthExpiredAt);
+
+  const { amount: energyAmount, expiredAt: _energyExpiredAt } = energy || {};
+  const energyExpiredAt = +new Date(_energyExpiredAt);
+
+  const UnfreezeBandwidth = useMemo(() => BigNumber(bandwidthAmount || 0), [bandwidthAmount]);
+  const canUnfreezeBandwidth = useMemo(
```

Comment:
that logic should exists in live-common.

we probably should have in live-common a unified function that tells me all the capabilities (canUnfreeze*, canClaimFunds, etc..)

Reply:
i'll make a PR for this it will be more like a formatter for unfreeze Data :+1: